apicast/src/proxy.lua

```diff
@@ -239,7 +241,44 @@ function _M:set_backend_upstream(service)
   ngx.ctx.backend_upstream = backend_upstream
 
   ngx.var.backend_endpoint = scheme .. '://backend_upstream' .. path
-  ngx.var.backend_host = service.backend.host or server or ngx.var.backend_host
+  ngx.var.backend_host = backend.host or server or ngx.var.backend_host
+end
+
+local function debug_headers(service, usage, credentials)
```

Comment:
The name is a bit confusing, as it suggests that it only has to do with the 3scale debug headers, while there are some lines which are an important part of the functionality.